### ISSUE TITLE
DeprecationWarning from `inspect.getargspec()`

### ISSUE THREAD

**Issue by frexvahi** [tensorflow/tensorflow#16152]
`inspect.getargspec` is deprecated in Python 3
https://docs.python.org/3/library/inspect.html#inspect.getargspec

I solved the problem in keras like this:
https://github.com/keras-team/keras/pull/7035

### System information
- Using tensorflow as a keras backend (keras 2.1.2)
- Linux Ubuntu 16.04
- installed from conda
- version 1.3.0
- python 3.6.4

### Describe the problem
We recently switched from theano to tensorflow and this warning message is filling up my test output.

### Source code / logs
```
/home/<name>/.conda/envs/<env>/lib/python3.6/site-packages/tensorflow/python/util/tf_inspect.py:45: DeprecationWarning:
  
  inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
```

**Comment by qmick** [comment]
It was fixed in a newer version, see https://github.com/tensorflow/tensorflow/blob/07e497b0ad3f0e102e167a50de2b543a4a238fde/tensorflow/python/util/tf_inspect.py#L62

**Comment by aselle** [comment]
@martinwicke, FYI
@frexvahi, would you be willing to submit a PR to fix this?

**Comment by frexvahi** [comment]
I could submit a PR, but getting someone to sign the corporate CLA would be harder.

**Comment by tensorflowbutler** [comment]
The original poster has replied to this issue after the stat:awaiting response label was applied.

**Comment by KarlJack47** [comment]
I fixed it on my end. I will try to submit a pull request today or tomorrow with the fix.

**Comment by martinwicke** [comment]
Can you say "Fixes #16152" in the description for your PR? That way this issue will be closed when your PR is submitted.

**Comment by KarlJack47** [comment]
A quick update. I believe the issue has been fixed in r1.7 since ever since I started using it I haven't had any of the warnings.

**Comment by martinwicke** [comment]
Thanks!

**Comment by pwaller** [comment]
I'm still hitting this, Tensorflow 1.7.0 and 1.8.0-rc1. Python 3.6.5. Afraid I don't have an easy reproducer to hand, will update if I get one.

**Comment by pwaller** [comment]
Reproducer below tested on 1.8.0-rc1. Was hard to make a reproducer because the warning did not show up unless I had `warnings.filterwarnings('error')`, which I use to make it easier to find the sources of warnings.

```python3
import tensorflow as tf

import warnings
warnings.filterwarnings('error')

tf.reduce_sum(tf.placeholder(tf.float64))
```

Full stack trace below.

<details>

```Python traceback
---------------------------------------------------------------------------
DeprecationWarning                        Traceback (most recent call last)
<ipython-input-1-1de048b23827> in <module>()
      4 warnings.filterwarnings('error')
      5 
----> 6 tf.reduce_sum(tf.placeholder(tf.float64))

~/Library/Python/3.6/lib/python/site-packages/tensorflow/python/util/deprecation.py in new_func(*args, **kwargs)
    403       if is_in_graph_mode.IS_IN_GRAPH_MODE() and _PRINT_DEPRECATION_WARNINGS:
    404         invalid_args = []
--> 405         named_args = tf_inspect.getcallargs(func, *args, **kwargs)
    406         for arg_name, spec in iter(deprecated_positions.items()):
    407           if (spec.position < len(args) and

~/Library/Python/3.6/lib/python/site-packages/tensorflow/python/util/tf_inspect.py in getcallargs(func, *positional, **named)
    107   argspec will be used.
    108   """
--> 109   argspec = getargspec(func)
    110   call_args = named.copy()
    111   this = getattr(func, 'im_self', None) or getattr(func, '__self__', None)

~/Library/Python/3.6/lib/python/site-packages/tensorflow/python/util/tf_inspect.py in getargspec(object)
     53   decorators, target = tf_decorator.unwrap(object)
     54   return next((d.decorator_argspec for d in decorators
---> 55                if d.decorator_argspec is not None), _inspect.getargspec(target))
     56 
     57 

/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py in getargspec(func)
   1069     warnings.warn("inspect.getargspec() is deprecated, "
   1070                   "use inspect.signature() or inspect.getfullargspec()",
-> 1071                   DeprecationWarning, stacklevel=2)
   1072     args, varargs, varkw, defaults, kwonlyargs, kwonlydefaults, ann = \
   1073         getfullargspec(func)

DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()


1
​
---------------------------------------------------------------------------
DeprecationWarning                        Traceback (most recent call last)
<ipython-input-4-8f069a824635> in <module>()
----> 1 tf.reduce_sum(tf.placeholder(tf.float64))

~/Library/Python/3.6/lib/python/site-packages/tensorflow/python/util/deprecation.py in new_func(*args, **kwargs)
    403       if is_in_graph_mode.IS_IN_GRAPH_MODE() and _PRINT_DEPRECATION_WARNINGS:
    404         invalid_args = []
--> 405         named_args = tf_inspect.getcallargs(func, *args, **kwargs)
    406         for arg_name, spec in iter(deprecated_positions.items()):
    407           if (spec.position < len(args) and

~/Library/Python/3.6/lib/python/site-packages/tensorflow/python/util/tf_inspect.py in getcallargs(func, *positional, **named)
    107   argspec will be used.
    108   """
--> 109   argspec = getargspec(func)
    110   call_args = named.copy()
    111   this = getattr(func, 'im_self', None) or getattr(func, '__self__', None)

~/Library/Python/3.6/lib/python/site-packages/tensorflow/python/util/tf_inspect.py in getargspec(object)
     53   decorators, target = tf_decorator.unwrap(object)
     54   return next((d.decorator_argspec for d in decorators
---> 55                if d.decorator_argspec is not None), _inspect.getargspec(target))
     56 
     57 

/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py in getargspec(func)
   1069     warnings.warn("inspect.getargspec() is deprecated, "
   1070                   "use inspect.signature() or inspect.getfullargspec()",
-> 1071                   DeprecationWarning, stacklevel=2)
   1072     args, varargs, varkw, defaults, kwonlyargs, kwonlydefaults, ann = \
   1073         getfullargspec(func)

DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
```

</details>

**Comment by CicoZhang** [comment]
Tensorflow 1.9.0-dev20180427
```
import tensorflow as tf
miniconda3/envs/kaggle/lib/python3.5/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
```

**Comment by sque** [comment]
I am still hitting this warning with the following versions:
 * 1.8.0
 * 1.9.0
 * 1.10.0
 * 1.10.1
 * 1.11.0-rc2

**Comment by EelcoHoogendoorn** [comment]
Had so many of these on 1.8 that i could not longer see my test output. Updated to 1.10 and now i only have a few dozen of these warnings left.

**Comment by joshburkart** [comment]
Yeah, still seeing this in 1.11.0-rc0:

```
/Users/josh/anaconda3/lib/python3.6/site-packages/tensorflow/python/util/tf_inspect.py:75: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
  return _inspect.getargspec(target)
/Users/josh/anaconda3/lib/python3.6/site-packages/tensorflow/python/util/tf_inspect.py:75: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
  return _inspect.getargspec(target)
...
```

Can we reopen?

**Comment by bwanglzu** [comment]
any updates?

**Comment by yongtang** [comment]
I think the issue has been resolved in #22517 now.

**Comment by NicoCoallier** [comment]
Still getting this issue with 1.11.0 and Python 3.6

**Comment by bwanglzu** [comment]
me too.

**Comment by HanxiSun** [comment]
me too

**Comment by Arvinds-ds** [comment]
I see it in 1.12 too on Python 3.6

**Comment by zqs01** [comment]
> I see it in 1.12 too on Python 3.6

the same to you

**Comment by wookayin** [comment]
Resolved in #22517, so I believe it will appear in the next release (TF 1.13 or 2.0).
